### DISCUSSION TITLE
expose all the "standard" HQL functions via the criteria API

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#5579] [Design Proposals]
I've already got two different requests for the `sql()` function in criteria queries, but really we need to make sure that *all* the standard functions are callable in criteria.

And also review the APIs to see that we're not missing all sorts of other new features of HQL (window functions, etc.)

**Comment by beikov** [comment]
Yes, we need to do that. Also see https://hibernate.atlassian.net/browse/HHH-15654

**Reply by gavinking** [reply]
Ah good, so there's an issue already.

**Comment by beikov** [comment]
The rest will be handled as part of https://hibernate.atlassian.net/browse/HHH-15868
